guides/bookshop-astro-guide/using-structures.mdx

```diff
@@ -0,0 +1,168 @@
+
+<comp.CodeBlock language="yaml" source="sample.bookshop.yml">
+```
+spec:
+  structures: #1
+    - content_blocks
+  label: My Component #2
+  description: A very simple component #3
+  icon: nature_people #4
+```
+<comp.Annotation number="1">
+  The `structures` key tells CloudCannon where this component can be added. In this example,
+  we are adding this component to the set of `content_blocks` structures.
+  In practice, this means that if a `content_blocks` key exists anywhere on your site,
+  for example, in the front matter of a page, CloudCannon will give editors the option to add
+  the `sample` component to that array or object.
+</comp.Annotation> <comp.Annotation number="2">
+  The `label` key is used as the main title of the component when selecting new components,
+  and browsing the existing components on a page.
+</comp.Annotation> <comp.Annotation number="3">
+  The `description` key can help provide context when selecting a new component in CloudCannon.
+</comp.Annotation> <comp.Annotation number="4">
+  The `icon` key is shown alongside existing components in CloudCannon, and when selecting new components.
+  This should be the attribute name of any [Material Icon](https://fonts.google.com/icons).
+</comp.Annotation></comp.CodeBlock>
+
+## The component blueprint
+
+The component `blueprint` is used to generate the CloudCannon Structure for your component. This object will be used as the initial data when the component is added to a page and represents all of the editable data that your component takes.
+
+If our `sample` component contained the following blueprint:
+
+<comp.CodeBlock language="yaml" source="sample.bookshop.yml">
+```
+blueprint:
+  text: "Hello World!"
+  color: "#034AD8"
+  image:
+    image_url: "/image.png"
+    image_alt: "My alternate text"
+```
+</comp.CodeBlock>
+
+Then creating a new `sample` component in CloudCannon would present us with the following data editor:
+
+<comp.DocsImage path="/documentation/static/bookshop-11ty/new-sample-component.png" alt="A data editor panel open in CloudCannon showing the initialized state of a new Sample component when added to the page" type="ui-snippet"/>
+
+Component blueprints contain some other powerful features for helping create your Structures. We'll see those soon in the [Nesting components](/documentation/guides/bookshop-eleventy-guide/nesting-components/) page of this guide.
+
+## Generating and using Structures
+
+The `npx @bookshop/generate` command that we set up when [configuring live editing](/documentation/guides/bookshop-eleventy-guide/configuring-live-editing/) will automatically create Structures from your Bookshop components and make them available to CloudCannon. This process happens as part of your site's build step on CloudCannon.
+
+The Structures to which that your component is assigned will determine where you can add it in CloudCannon. For a `sample` component tagged as:
```

Comment:
Confusing sentence, perhaps:

"The Structures that your component is assigned to will determine where you can add it in CloudCannon."

or

"The Structures listed in your component's spec will determine where you can add that component in CloudCannon"